### ISSUE TITLE
User story - User Profile Search

### ISSUE THREAD

**Issue by stephendawsondev** [stephendawsondev/StackPortfolio#17]
### If applicable, add a related Epic.

Epic: #14 , #42 

### User Story

As a **user**, I can **search user profiles** so that I can **find friends, colleagues or potential candidates to connect with**.

### Acceptance Criteria

- The user should be able to search for users by name (at least)
- Search results should be displayed in a logical order
- There should be a search results page, ajax search or both

### Please ensure you've completed these tasks before marking the issue as done.

- [x] Reviewed the user story for clarity and completeness.
- [x] Identified and linked related issues or epics.
- [x] Created or updated the necessary documentation.
- [ ] Written unit and/or integration tests for functionality.
- [ ] All code changes have been manually tested to ensure they meet the acceptance criteria.
- [x] Updated the issue with any relevant notes or feedback (to date).
- [x] Merged into the main branch and deployed to production (if applicable).

### Please add any further tasks that need to be completed before the issue can be marked as done.

- [x] Create a search page or search box for user input
- [x] Query the database for users whose names match the input pattern
- [x] Filter the users and output their details using cards

**Comment by stephendawsondev** [comment]
This was completed in #96
